src/main/java/codeu/controller/ConversationServlet.java

```diff
@@ -70,8 +72,30 @@ void setConversationStore(ConversationStore conversationStore) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
+    String username = (String) request.getSession().getAttribute("user");
+    User user = userStore.getUser(username);
+
     List<Conversation> conversations = conversationStore.getAllConversations();
-    request.setAttribute("conversations", conversations);
+    List<Conversation> toShow = new ArrayList<Conversation>(conversations);
+
+    if (user == null) { //if no user, show no conversations
+      toShow.clear();
+    } else{
+      toShow = conversationStore.getAllConversations();
+    }
+    for (Iterator<Conversation> iterator = toShow.iterator(); iterator.hasNext();) {
```

Comment:
Instead of removing from this list, I'd recommend creating a new list and only adding results with the right member to it.

so there'd be a memberConversations.add instead.